### PR TITLE
Add `irc_prejoin_commands`

### PR DESCRIPTION
**Pull request by llmII**
Allows for the bot to send raw messages to the IRC server for each connection spawned. Useful for networks that forcefully join newly connected users to a channel they have no intention of remaining in.

### CODE REVIEW

File: bridge/irc_listener.go [hunk 1]
```diff
@@ -164,6 +164,11 @@ func (i *ircListener) OnWelcome(e *irc.Event) {
 		i.Privmsgf("nickserv", "identify %s", identify)
 	}
 
+	//execute peform
```
Comment (qaisjp) [inline bridge/irc_listener.go:167]:
```suggestion
	// Execute global perform commands
```

**Review by qaisjp — CHANGES_REQUESTED** [review]
Thanks, this looks good.

- Probably want to use `SendRaw` instead of `SendRawf`
- `README.md` has a list of settings and needs to be updated (for the other PRs too)
- `config.yml` is missing a trailing newline

File: bridge/irc_connection.go [hunk 2]
```diff
@@ -31,9 +31,15 @@ type ircConnection struct {
 }
 
 func (i *ircConnection) OnWelcome(e *irc.Event) {
-	i.JoinChannels()
 	i.innerCon.SendRawf("MODE %s +D", i.innerCon.GetNick())
 
+	// execute perform second
+	for _, com := range i.manager.bridge.Config.IRCGlobalPerform {
+		i.innerCon.SendRawf(com)
```
Comment (qaisjp) [inline bridge/irc_connection.go:38]:
```suggestion
		i.innerCon.SendRaw(com)
```

File: config.yml [hunk 3]
```diff
@@ -25,3 +25,6 @@ no_tls: false
 debug: false
 webhook_prefix: "(auto-test)"
 # simple: true
+# global_perform:
+#   - PART #forced-to-join-test-channel
+#   - PRIVMSG Nick :msg
```
Comment (qaisjp) [inline config.yml:30]:
```suggestion
#   - PRIVMSG Nick :msg

```

File: bridge/irc_listener.go [hunk 4]
```diff
@@ -164,6 +164,11 @@ func (i *ircListener) OnWelcome(e *irc.Event) {
...
+	for _, com := range i.bridge.Config.IRCGlobalPerform {
+		i.SendRawf(com)
```
Comment (qaisjp) [inline bridge/irc_listener.go:169]:
```suggestion
		i.SendRaw(com)
```

**Comment by llmII** [comment]
> * `README.md` has a list of settings and needs to be updated (for the other PRs too)

I definitely need to do that. I'll tack on a commit soon for `README.md` where appropriate.

File: bridge/irc_connection.go [hunk 5]
```diff
@@ -31,9 +31,15 @@ type ircConnection struct {
 }
 
 func (i *ircConnection) OnWelcome(e *irc.Event) {
-	i.JoinChannels()
 	i.innerCon.SendRawf("MODE %s +D", i.innerCon.GetNick())
 
+	// execute perform second
+	for _, com := range i.manager.bridge.Config.IRCGlobalPerform {
```
Comment (qaisjp) [inline bridge/irc_connection.go:37]:
```suggestion
	for _, com := range i.manager.bridge.Config.IRCPrejoinCommands {
```

File: bridge/irc_listener.go [hunk 6]
```diff
@@ -180,6 +180,11 @@ func (i *ircListener) OnWelcome(e *irc.Event) {
 		i.Privmsgf("nickserv", "identify %s", identify)
 	}
 
+	// Execute global perform commands
+	for _, com := range i.bridge.Config.IRCGlobalPerform {
```
Comment (qaisjp) [inline bridge/irc_listener.go:184]:
```suggestion
	for _, com := range i.bridge.Config.IRCPrejoinCommands {
```

**Review by qaisjp — APPROVED** [review]
Thank you! This works really well. I changed the setting name.

**Pull request merged**